File: library/src/main/java/com/yarolegovich/slidingrootnav/transform/YTranslationTransformation.java
package com.yarolegovich.slidingrootnav.transform;

import android.view.View;

import com.yarolegovich.slidingrootnav.util.SideNavUtils;

/**
 * Created by yarolegovich on 25.03.2017.
 */

public class YTranslationTransformation implements RootTransformation {

    private static final float START_TRANSLATION = 0f;

    private final float endTranslation;

    public YTranslationTransformation(float endTranslation) {
        this.endTranslation = endTranslation;
    }

    @Override
    public void transform(float dragProgress, View rootView) {
        float translation = SideNavUtils.evaluate(dragProgress, START_TRANSLATION, endTranslation);
        rootView.setTranslationY(translation);
    }
}

File: library/src/main/java/com/yarolegovich/slidingrootnav/transform/YTranslationTransformation.kt
package com.yarolegovich.slidingrootnav.transform

import android.view.View

import com.yarolegovich.slidingrootnav.util.SideNavUtils

/**
 * Created by yarolegovich on 25.03.2017.
 * Edited by Mehdi on 27.02.2019
 */

class YTranslationTransformation(private val endTranslation: Float) : RootTransformation {

    override fun transform(dragProgress: Float, rootView: View) {
        val translation = SideNavUtils.evaluate(dragProgress
                , START_TRANSLATION
                , endTranslation)
        rootView.translationY = translation
    }

    companion object {

        private val START_TRANSLATION = 0f
    }
}
